### PR TITLE
GitHub 61 add dsp primitives

### PR DESCRIPTION
**Pull request by rjdiez**
Added dsp primitives (issue #61). 
Files: 
- dsp.py (candidates folder)
- mlprimitives.candidates.dsp.Moments.json (jsons folder)
- mlprimitives.candidates.dsp.SpectralMask.json (jsons folder)

### CODE REVIEW

File: mlprimitives/candidates/dsp.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+            anomalies[i] = 0
+            for m in range(0, int(window_length / 2) - 1):
+                if ((sig_freq[m] > mask_top[m]) or (sig_freq[m] < mask_bottom[m])):
+                    anomalies[i] = 1
+                    break
+        self.time_produce = time.time() - start
+        print('DONE DETECTING ANOMALIES - SPECTRAL MASK')
+        print('Time consumed detecting anomalies: ', self.time_produce)
+
+        return anomalies  # , self.time_freq_compare
```
Comment (bryanfarris) [inline mlprimitives/candidates/dsp.py]: Remove the commented code.

File: mlprimitives/candidates/dsp.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        for i in range(0, len(signal) - window_length - 1):
+            sig_freq = np.abs(np.fft.rfft(signal[i:i + window_length] * window)) / window_length
+            anomalies[i] = 0
+            for m in range(0, int(window_length / 2) - 1):
+                if ((sig_freq[m] > mask_top[m]) or (sig_freq[m] < mask_bottom[m])):
+                    anomalies[i] = 1
+                    break
+        self.time_produce = time.time() - start
+        print('DONE DETECTING ANOMALIES - SPECTRAL MASK')
+        print('Time consumed detecting anomalies: ', self.time_produce)
```
Comment (bryanfarris) [inline mlprimitives/candidates/dsp.py]: @csala Should we be printing the anomalies?

File: mlprimitives/jsons/dsp.stdDevFreqAnalysis.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "documentation": "",
+    "description": "Anomalies detection in satellite telemetry data using the standard deviation of the frequency components calculated using time-dependent Fast Fourier Transform",
+    "modalities": [],
+    "primitive": "mlprimitives.dsp.freqAnalysis",
+    "fit": {
+        "method": "fitFreqStdDev",        
+        "args": [
+            {
+                "name": "X",
+                "description": "training signal",
```
Comment (bryanfarris) [inline mlprimitives/jsons/dsp.stdDevFreqAnalysis.json]: Please capitalize, "Training signal"  to make it more professional

File: mlprimitives/jsons/dsp.stdDevFreqAnalysis.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                "name": "X",
+                "description": "telemetry data vector",
+                "type": "ndarray"
+            }
+        ],
+        "output": [
+            {
+                
+                "name": "anomalies",
+                "descrtiption": "data vector consisting of anomalies detected in the input vector",
```
Comment (bryanfarris) [inline mlprimitives/jsons/dsp.stdDevFreqAnalysis.json]: 1.  Description is misspelled.
2.  Please capitalize "Data vector...."  to make it more professional.  Please do this to all descriptions, I won't comment on each individual one.

File: mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                "name": "X",
+                "description": "Telemetry data vector",
+                "type": "ndarray"
+            }
+        ],
+        "output": [
+            {
+                
+                "name": "anomalies",
+                "descrtiption": "Data vector consisting of the anomalies detected in the telemetry data",
```
Comment (bryanfarris) [inline mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json]: Description is misspelled

File: mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                "type": "ndarray"
+            }
+        ]
+    },
+
+    "hyperparameters": {
+        "tunable": {
+            "gain": {
+                "description": "Multiplication factor used in the comparisson between the moments calculated using the training data and the moments of the telemetry data vector",
```
Comment (bryanfarris) [inline mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json]: "comparison" is misspelled.  Please fix other instances of "comparison" below as well.

File: mlprimitives/jsons/dsp.stdDevFreqAnalysis.json (hunk 4 above)
Comment (rjdiez) [reply mlprimitives/jsons/dsp.stdDevFreqAnalysis.json]: This JSON file is not used anymore, I'll remove it.

File: mlprimitives/jsons/dsp.stdDevFreqAnalysis.json (hunk 3 above)
Comment (rjdiez) [reply mlprimitives/jsons/dsp.stdDevFreqAnalysis.json]: This JSON file is not used anymore, I'll remove it.

File: mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json (hunk 5 above)
Comment (rjdiez) [reply mlprimitives/jsons/mlprimitives.candidates.dsp.Moments.json]: This JSON file is not used anymore, I'll remove it.

**Pull request closed without merging**